packages/relay-compiler/core/GraphQLIRPrinter.js

```diff
@@ -304,13 +304,14 @@ function printLiteral(value: mixed, type: ?GraphQLInputType): string {
     type = type.ofType;
   }
   if (type instanceof GraphQLEnumType) {
+    const result = type.serialize(value);
     invariant(
-      typeof value === 'string',
-      'GraphQLIRPrinter: Expected value of type %s to be a string, got `%s`.',
+      typeof result === 'string',
+      'GraphQLIRPrinter: Expected value of type %s to be a valid enum value, got `%s`.',
       type.name,
       value,
     );
-    return value;
+    return result;
```

Comment:
I can't remember if GraphQLInputObject and GraphQLList also support `serialize()`, if they do can we just use the approach here for all types (ie replace the body of `printLiteral` w the body of the enum case)? Even if lists/objects don't work, we could do this for all enums and scalars.

Reply:
Only scalars and enums have serialize.